commit fa8e94c9caadf1f53746333220039e3c0ab4f9c4
Author: Marius Vollmer <mvollmer@redhat.com>
Date:   Mon Jul 4 11:45:18 2016 +0300

    ISCSI: Add SessionsSupported property
    
    So that clients like Cockpit can adapt accordingly.

diff --git a/modules/iscsi/data/org.freedesktop.UDisks2.iscsi.xml b/modules/iscsi/data/org.freedesktop.UDisks2.iscsi.xml
index 7b18f0b..d6bb4d6 100644
--- a/modules/iscsi/data/org.freedesktop.UDisks2.iscsi.xml
+++ b/modules/iscsi/data/org.freedesktop.UDisks2.iscsi.xml
@@ -181,6 +181,13 @@
       <arg name="iface" direction="in" type="s"/>
       <arg name="options" type="a{sv}" direction="in"/>
     </method>
+
+    <!--
+        SessionsSupported: Whether or not this version of Storaged
+        supports ISCSI.Session objects.
+    -->
+    <property name="SessionsSupported" type="b" access="read"/>
+
   </interface>
 
   <!--
diff --git a/modules/iscsi/udiskslinuxmanageriscsiinitiator.c b/modules/iscsi/udiskslinuxmanageriscsiinitiator.c
index e93c0a6..db9f2f8 100644
--- a/modules/iscsi/udiskslinuxmanageriscsiinitiator.c
+++ b/modules/iscsi/udiskslinuxmanageriscsiinitiator.c
@@ -184,6 +184,10 @@ udisks_linux_manager_iscsi_initiator_init (UDisksLinuxManagerISCSIInitiator *man
 
   g_dbus_interface_skeleton_set_flags (G_DBUS_INTERFACE_SKELETON (manager),
                                        G_DBUS_INTERFACE_SKELETON_FLAGS_HANDLE_METHOD_INVOCATIONS_IN_THREAD);
+#ifdef HAVE_LIBISCSI_GET_SESSION_INFOS
+  udisks_manager_iscsi_initiator_set_sessions_supported (UDISKS_MANAGER_ISCSI_INITIATOR (manager),
+                                                         TRUE);
+#endif
 }
 
 /**
